opal/mca/threads/pthreads/threads_pthreads_module.c

```diff
@@ -53,50 +42,55 @@ static void opal_thread_construct(opal_thread_t *t)
     t->t_handle = (pthread_t) -1;
 }
 
-int opal_thread_start(opal_thread_t *t)
-{
-    int rc;
+OBJ_CLASS_INSTANCE(opal_thread_t,
+                   opal_object_t,
+                   opal_thread_construct, NULL);
+     
 
-    if (OPAL_ENABLE_DEBUG) {
-        if (NULL == t->t_run || t->t_handle != (pthread_t) -1) {
-            return OPAL_ERR_BAD_PARAM;
-        }
-    }
 
-    rc = pthread_create(&t->t_handle, NULL, (void*(*)(void*)) t->t_run, t);
+opal_thread_t *opal_thread_get_self(void)
+{
+    opal_thread_t *t = OBJ_NEW(opal_thread_t);
+    t->t_handle = pthread_self();
+    return t;
+}
 
-    return (rc == 0) ? OPAL_SUCCESS : OPAL_ERROR;
+bool opal_thread_self_compare(opal_thread_t *t)
+{
+    return t->t_handle == pthread_self();
 }
 
+int sync_wait_mt(void *p) {
```

Comment:
We have very few style rules in Open MPI.  But one of them is that the opening `{` for functions must be on a new line.  😄